### DISCUSSION TITLE
How to work with relative links in a site in a subpath (base)

### DISCUSSION THREAD

**Discussion by joesonw** [ElMassimo/iles#254] [Q&A]
I have `siteUrl` set to something like `http://example.com/abc`, it'll always redirect to `http://example.com/abc/`, and it gives inconsistent link behaviours.  Say if I have a href link to `http://example.com/abc/123`, it will actually go to `http://example.com/abc/abc/123` in dev environment, and no such problem in built files.

So is there a way to remove trailing slashes? I tried to set every route to `strict` in `enhanceApp`, but it did not help.

**Comment by ElMassimo** [comment]
Hi Joe!

Didn't notice it the first time, but if your `siteUrl` is `http://example.com/abc`, so `base` will be `/abc/`.

If you use the `href` provided in `usePage` it should [handle the `base` automatically](https://github.com/ElMassimo/iles/blob/main/packages/pages/src/api.ts#L68) for you.

Managing relative URLs in a site that is hosted in a sub-path is tricky, but now we can discard other causes, such as trailing slash.

---

~~Please check the [`prettyUrls`](https://iles.pages.dev/config#prettyurls) setting, though it's enabled by default.~~

**Reply by joesonw** [reply]
I explicityly enabled it, and it still behaves the same. It's only behaving this way in dev environment though.

**Reply by joesonw** [reply]
There seems to be some kind of interceptor running, it only appends prefixes in dev mode. So if I get rid of the manual added prefix in url, it would not in build mode.

**Reply by ElMassimo** [reply]
Try running [`vite-plugin-inspect`](https://github.com/antfu/vite-plugin-inspect) to see if the page file is modified in an unexpected way.

> Feel free to provide a repo, or the page file that defines those routes.

**Reply by joesonw** [reply]
> Try running [`vite-plugin-inspect`](https://github.com/antfu/vite-plugin-inspect) to see if the page file is modified in an unexpected way.
> 
> > Feel free to provide a repo, or the page file that defines those routes.

There were no changes. And by inspecting the elements, they urls are still `http://example.com/abc/123`, but once I click it, it goes to `http://example.com/abc/abc/123`

**Reply by ElMassimo** [reply]
Didn't notice it the first time, but if your `siteUrl` is `http://example.com/abc`, so `base` will be `/abc/`.

If you use the `href` provided in `usePage` it should [handle the `base` automatically](https://github.com/ElMassimo/iles/blob/main/packages/pages/src/api.ts#L68) for you.

Managing relative URLs in a site that is hosted in a sub-path is tricky, but now we can discard other causes, such as trailing slash.

**Reply by joesonw** [reply]
> Didn't notice it the first time, but if your `siteUrl` is `http://example.com/abc`, so `base` will be `/abc/`.
> 
> If you use the `href` provided in `usePage` it should [handle the `base` automatically](https://github.com/ElMassimo/iles/blob/main/packages/pages/src/api.ts#L68) for you.
> 
> Managing relative URLs in a site that is hosted in a sub-path is tricky, but now we can discard other causes, such as trailing slash.

Yeah, I did get the `href` part through some other way. But like I said, a link of `http://example.com/abc/123` will actually go to `http://example.com/abc/abc/123`. But if you open it in a new tab,  then no problem. There is definitely something going on with the router, I just don't know what.

**Reply by ElMassimo** [reply]
In development [`vue-router` is configured to use provided `base`](https://github.com/ElMassimo/iles/blob/main/packages/iles/src/client/app/index.ts#L46).

It could be that [this script](https://github.com/ElMassimo/iles/blob/main/packages/iles/src/client/app/composables/routerLinks.ts#L30), meant to provide client-side navigation during development is not taking the `base` into account?
